### DISCUSSION TITLE
Fastapi - using Core and ORM together (in async mode)

### DISCUSSION THREAD

**Discussion by mfrey777** [sqlalchemy/sqlalchemy#6542] [Usage Questions]
Hello,
I am in the process of migrating/redelloping a Flask based App.

I want to migrate it to FastApi (and use async as much as possible) and am also looking at replacing my direct database connection with sqlalchemy.

The challenge is that my app has 80 % of tables that would perfectly fit into the ORM (that can be modelled as object with relationships) and about 20% of tables that are dynamic and where complex SQL queries are required for data analysis & calculations (aggregating data according to hierarchies, creating/updating these hierarchies).

My idea is to use the ORM and Core parts together (but of course not updating the same tables from both). 

So I plan to use Core ( async with engine.connect() as conn:) and ORM (async with async_session() as session: / async with session.begin(): toghether 

Has anybody tried that before, Can this approach work is that a bad idea (if yes, I would of course welcome an alternative suggestion)

**Comment by CaselIT** [comment]
I'm not using async, but I do use both core and orm for a similar reason as yours. Usually I use a session, since it works with both and if I need a connection there is always the `session.connection()` method to get the connection used by the session

**Reply by mfrey777** [reply]
Thank you for the feed-back.
The same logic seems to also work with async. I can use the connection to execute a Core select or plain SQL
conn = await session.connection()
        resultset = await conn.execute(select(Record).order_by(Record.id).where(Record.id < 100))
        resultset2 = await conn.execute(text('SELECT * FROM Records WHERE id < 100 ORDER BY id'))
I of coures have to assess the peformance of that approach when I do performance tests later on in the development

**Reply by CaselIT** [reply]
You should also be able to run that statements directly on the session getting the same result back (if Recod is a table and not a mapped class)

**Reply by mfrey777** [reply]
Thank you, that indeed works :-)
